controllers/volumereplicationgroup_controller_test.go

```diff
@@ -935,7 +935,22 @@ func (v *vrgTest) waitForVolRepPromotion(vrNamespacedName types.NamespacedName,
 		// as of now name of VolumeReplication resource created by the VolumeReplicationGroup
 		// is same as the pvc that it replicates. When that changes this has to be changed to
 		// use the right name to get the appropriate protected PVC condition from VRG status.
-		dataReadyCondition := checkConditions(vrg.Status.ProtectedPVCs[updatedVolRep.Name].Conditions,
+		var protectedPVC *ramendrv1alpha1.ProtectedPVC
+		for index := range vrg.Status.ProtectedPVCs {
+			curPVC := &vrg.Status.ProtectedPVCs[index]
+			if curPVC.Name == updatedVolRep.Name {
+				protectedPVC = curPVC
+
+				break
+			}
+		}
```

Comment:
(nit) Perhaps the loop here and the loop in 1912 can be replaced with a single function.